shoop/front/basket/storage.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+import six
+import abc
+from django.conf import settings
+from django.contrib.auth.models import AnonymousUser
+from django.core.exceptions import ImproperlyConfigured
+from shoop.core.models.contacts import AnonymousContact
+from shoop.front.models import StoredBasket
+from shoop.utils.importing import cached_load
+
+
+class BasketStorage(six.with_metaclass(abc.ABCMeta)):
+    @abc.abstractmethod
+    def load(self, basket):  # pragma: no cover
+        """
+        Load the given basket's data dictionary from the storage.
+
+        :type basket: shoop.front.basket.objects.BaseBasket
+        """
+        return {}
+
+    @abc.abstractmethod
+    def save(self, basket, data):  # pragma: no cover
+        """
+        Save the given data dictionary into the storage for the given basket.
+
+        :type basket: shoop.front.basket.objects.BaseBasket
+        """
+        pass
+
+    @abc.abstractmethod
+    def delete(self, basket):  # pragma: no cover
+        """
+        Delete the basket from storage.
+
+        :type basket: shoop.front.basket.objects.BaseBasket
+        """
+        pass
+
+    def finalize(self, basket):  # pragma: no cover
+        """
+        Mark the basket as "finalized" (i.e. completed) in the storage.
+
+        The actual semantics of what finalization does are up to each backend.
+
+        :type basket: shoop.front.basket.objects.BaseBasket
+        """
+        self.delete()
+
+
+class DirectSessionBasketStorage(BasketStorage):
+    def __init__(self):
+        if settings.SESSION_SERIALIZER == "django.contrib.sessions.serializers.JSONSerializer":  # pragma: no cover
```

Comment:
@suutari-ai: the newest commit replaces pickling (in the database anyway) with tagged JSON. What do you think?